opendevin/README.md

```diff
@@ -18,15 +19,32 @@ The key classes in OpenDevin are:
     * SessionManager: keeps a list of active sessions, and ensures requests are routed to the correct Session
 
 ## Control Flow
+Here's the basic loop (in pseudocode) that drives agents.
+```python
+while True:
+  prompt = agent.generate_prompt(state)
+  response = llm.completion(prompt)
+  action = agent.parse_response(response)
+  observation = runtime.run(action)
+  state = state.update(action, observation)
+```
+
+In reality, most of this is achieved through message passing, via the EventStream.
 The EventStream serves as the backbone for all communication in OpenDevin.
 
 ```mermaid
 flowchart LR
   Agent--Actions-->AgentController
-  AgentController--Observations-->Agent
+  AgentController--State-->Agent
   AgentController--Actions-->EventStream
   EventStream--Observations-->AgentController
   Runtime--Observations-->EventStream
   EventStream--Actions-->Runtime
   Frontend--Actions-->EventStream
 ```
+
+## Runtime
+The Runtime class is abstract, and has a few different implementations:
+* We have a LocalRuntime, which runs commands and edits files directly on the user's machine
```

Comment:
Nit: we should lint markdown. I'd add a blank line above each list.

Reply:
Done!